### DISCUSSION TITLE
Feature request: sampling `LIST(...)` aggregate

### DISCUSSION THREAD

**Discussion by zenazn** [duckdb/duckdb#11609] [Ideas]
I'd love to be able to write the following query:

```sql
SELECT country, LIST(city USING SAMPLE reservoir(10)) AS examples FROM cities GROUP BY 1;

-- syntax alternative
SELECT country, LIST_SAMPLE(city, 10) AS examples FROM cities GROUP BY 1;
```

This is helpful when exploring and summarizing data

In case it's helpful, here's [clickhouse's `groupArraySample`](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/grouparraysample), which I believe provides similar functionality

**Comment by soerenwolfers** [comment]
Alternatively, the existing `USING SAMPLE` clause could accept a `WITHIN` sub clause. (I believe in many cases you actually want the rows without the list aggregation. Of course the two are equivalent via `unnest` and the `list` aggregate)
